db/migrate/20200721135726_move_calculators_preferences_outside_spree_namespace.rb

```diff
@@ -0,0 +1,23 @@
+# As we moved the calculators outside the Spree namespace in migration MoveAllCalculatorsOutsideTheSpreeNamespace
+#   We need to move their preferences too (currency, value, etc), otherwise they are not used
+class MoveCalculatorsPreferencesOutsideSpreeNamespace < ActiveRecord::Migration
+  def up
+    replace_preferences_key("/spree/calculator", "/calculator")
+  end
+
+  def down
+    replace_preferences_key("/calculator", "/spree/calculator")
+  end
+
+  private
+
+  def replace_preferences_key(from_pattern, to_pattern)
+    updated_pref_key = "replace( pref.key, '" + from_pattern + "', '" + to_pattern + "')"
+    Spree::Preference.connection.execute(
+      "UPDATE spree_preferences pref SET key = " + updated_pref_key +
+      " WHERE pref.key like '" + from_pattern + "%'" +
+      "   AND NOT EXISTS (SELECT 1 FROM spree_preferences existing_pref
+                          WHERE existing_pref.key = " + updated_pref_key + ")"
```

Comment:
There's blank spaces either side of the string in these keys?

Reply:
why do you ask? no blank spaces, just more characters like the name of the fee (flat_rate for example), that's why we need the %.

Reply:
maybe you are looking at updated_pref_key and thinking it's a string? updated_pref_key is a call to the replace function and so it can have spaces before and after.